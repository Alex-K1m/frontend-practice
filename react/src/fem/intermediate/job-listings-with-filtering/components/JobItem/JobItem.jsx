import cn from 'classnames';
import { nanoid } from 'nanoid';
import React from 'react';

import logo from '../../images/photosnap.svg';
import Chip from '../Chip';
import Logo from '../Logo';
import Tag from '../Tag';
import styles from './JobItem.module.scss';

const JobItem = () => {
  return (
    <div className={cn(styles.job, styles.jobHighlighted)}>
      <div className={styles.logo}>
        <Logo src={logo} alt='Photosnap logo' />
      </div>
      <div className={styles.body}>
        <div className={styles.heading}>
          <h2>Photosnap</h2>
          <Chip>New!</Chip>
          <Chip color='dark'>Featured</Chip>
        </div>
        <h3 className={styles.position}>Senior Frontend Developer</h3>
        <div className={styles.details}>
          <span>1d ago</span>
          <span>Full Time</span>
          <span>USA only</span>
        </div>
      </div>
      <hr className={styles.hr} />
      <ul className={styles.tagList}>
        {['Frontend', 'Senior', 'HTML', 'CSS', 'JavaScript'].map((label) => (
          <li className={styles.tagListItem} key={nanoid()}>
            <Tag onClick={() => console.log(`clicked ${label}`)}>{label}</Tag>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default JobItem;
