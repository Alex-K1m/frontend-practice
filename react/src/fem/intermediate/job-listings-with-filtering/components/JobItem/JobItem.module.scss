@use '../colors' as c;

$logo-size: 4.8rem;
$padding: 2.4rem;

.job {
  padding: $padding;
  background-color: white;
  border-radius: 0.5rem;
  margin-top: $logo-size / 2;

  &Highlighted {
    background-image: linear-gradient(
      to right,
      c.$primary 0.5rem,
      transparent 0.5rem,
    );
  }
}

.logo {
  width: $logo-size;
  height: $logo-size;
  margin-top: -$logo-size / 2 - $padding;
}

.body {
  margin-top: 0.8rem;
}

.heading {
  font: 700 1.3rem Spartan, sans-serif;
  color: c.$primary;
  display: flex;
  align-items: center;
  flex-flow: wrap;
  margin: -0.2rem -0.4rem;

  > :first-child {
    margin: 0.2rem 1.2rem 0.2rem 0.4rem;
  }

  > :not(:first-child) {
    margin: 0.2rem 0.4rem;
  }
}

.position {
  font: 700 1.5rem/160% Spartan, sans-serif;
  color: c.$text;
  margin-top: 0.8rem;
}

.details {
  font: 500 1.3rem/185% Spartan, sans-serif;
  letter-spacing: -0.0077em;
  color: c.$text-secondary;
  margin-top: 0.8rem;

  > :not(:first-child) {
    &::before {
      content: '';
      display: inline-block;
      margin: 0 0.8rem;
      width: 0.4rem;
      height: 0.4rem;
      border-radius: 50%;
      background-color: c.$text-secondary;
      vertical-align: middle;
    }
  }
}

.hr {
  border: none;
  height: 0.1rem;
  background-color: c.$text-secondary;
  width: 100%;
  margin: 1.6rem 0;
}

.tagList {
  margin: -0.8rem;
  display: flex;
  flex-flow: wrap;
}

.tagListItem {
  margin: 0.8rem;
}
