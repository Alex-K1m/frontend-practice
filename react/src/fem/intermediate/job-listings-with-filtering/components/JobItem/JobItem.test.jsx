import React from 'react';
import renderer from 'react-test-renderer';

import JobItem from './JobItem';

describe('JobItem', () => {
  // const mock = jest.fn();

  it('renders correctly', () => {
    const tree = renderer.create(<JobItem />);
    const json = tree.toJSON();
    // const { root } = tree;

    expect(json).toMatchSnapshot();
    // expect(root.findByProps({ className: 'tag' }).children).toEqual([tagName]);
  });
});
