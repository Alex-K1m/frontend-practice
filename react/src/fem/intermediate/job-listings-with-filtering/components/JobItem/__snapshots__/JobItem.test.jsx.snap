// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem renders correctly 1`] = `
<div
  className="job jobHighlighted"
>
  <div
    className="logo"
  >
    <img
      alt="Photosnap logo"
      className="logo"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="body"
  >
    <div
      className="heading"
    >
      <h2>
        Photosnap
      </h2>
      <div
        className="chip"
      >
        <span
          className="label"
        >
          New!
        </span>
      </div>
      <div
        className="chip dark"
      >
        <span
          className="label"
        >
          Featured
        </span>
      </div>
    </div>
    <h3
      className="position"
    >
      Senior Frontend Developer
    </h3>
    <div
      className="details"
    >
      <span>
        1d ago
      </span>
      <span>
        Full Time
      </span>
      <span>
        USA only
      </span>
    </div>
  </div>
  <hr
    className="hr"
  />
  <ul
    className="tagList"
  >
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <button
          className="tag btn"
          onClick={[Function]}
          type="button"
        >
          Frontend
        </button>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <button
          className="tag btn"
          onClick={[Function]}
          type="button"
        >
          Senior
        </button>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <button
          className="tag btn"
          onClick={[Function]}
          type="button"
        >
          HTML
        </button>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <button
          className="tag btn"
          onClick={[Function]}
          type="button"
        >
          CSS
        </button>
      </div>
    </li>
    <li
      className="tagListItem"
    >
      <div
        className="wrapper"
      >
        <button
          className="tag btn"
          onClick={[Function]}
          type="button"
        >
          JavaScript
        </button>
      </div>
    </li>
  </ul>
</div>
`;
